packages/shared/src/types/clerk/types.ts

```diff
@@ -19,3 +19,17 @@ export type UserPublicMetadata = {
   servicesHashes?: IServicesHashes;
   jobTitle?: JobTitleEnum;
 };
+
+/**
+ * Unsafe metadata can be updated from the frontend directly
+ */
+export type UserUnsafeMetadata = {
+  v3DashboardOptInStatus?: V3DashboardOptInStatusEnum;
+};
+
+export enum V3DashboardOptInStatusEnum {
```

Comment:
💡 I'd keep all this state into a single variable in clerk metadata:

`dashboard_version: 'v1' | 'v2' | null | undefined.

V1 is opt-out, V2 is opt-in, null is dismissed or never opted-in.

Reply:
This is inconclusive because user always uses one of the dashboards, either `v1` or `v2`.

The naming suggests I can use that variable to show something deterministically based on dashboard version, which is not true because by default if user didn't interacted with the opt-in widget he would have no `dashboard_version`, therefore `undefined`. Thats why I think its better to tie the variable name directly to the opt-in operation not the dashboard version.